websites/F/Free Web Novel/presence.ts

```diff
@@ -0,0 +1,189 @@
+					url: location.href
+				}
+			];
+			break;
+		}
+		case pathname === "/search/": {
+			presenceData.details = "Searching for a novel";
+			presenceData.state = `"${
+				document.querySelector("body > div.main > div.wp > div.m-t1 > em.e2")
+					.textContent
+			}"`;
+			break;
+		}
+		case pathname === "/contact.html": {
+			presenceData.details = "Viewing the contact page";
+			break;
+		}
+		case pathname === "/privacy-terms-of-use.html": {
+			presenceData.details = "Reading the terms of use";
+			break;
+		}
+		case privacy: {
+			presenceData.largeImageKey = "incognito";
+			if (
+				(pathnames.includes(pathname) ||
+					document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) &&
+				showReading
+			)
+				presenceData.details = "Browsing...";
+			else if (
+				document.querySelector<HTMLAnchorElement>(
+					'[title="Read Next chapter"]'
+				) &&
+				showReading
+			) {
+				presenceData.details = "Reading...";
+				presenceData.smallImageKey = "open";
+			}
+			break;
+		}
+		case showLogo: {
+			presenceData.largeImageKey = "fwn_1024";
+			delete presenceData.smallImageKey;
+			delete presenceData.details;
+			delete presenceData.state;
+			delete presenceData.buttons;
+			break;
+		}
+		default: {
+			if (document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) {
+				if (showCover) {
+					presenceData.largeImageKey = `${
+						document.querySelector<HTMLImageElement>(
+							"body > div.main > div > div > div.col-content > div.m-info > div.m-book1 > div.m-imgtxt > div.pic > img"
+						).src
+					}`;
+				} else presenceData.largeImageKey = "nocover";
+				if (showReading) presenceData.details = "Viewing a novel";
+				if (showBook) presenceData.state = document.title.split("-")[0];
+				presenceData.buttons = [
+					{
+						label: "View Novel",
+						url: location.href
+					}
+				];
+			} else if (
+				document.querySelector<HTMLAnchorElement>('[title="Read Next chapter"]')
+			) {
+				if (showCover) {
+					presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+						"head > meta:nth-child(4)"
+					).content;
+				} else presenceData.largeImageKey = "nocover";
+				if (showBook) {
+					presenceData.details = `Reading ${document.title.split("-")[0]}`;
+					presenceData.state =
+						document.querySelector<HTMLSpanElement>(
+							'[class="chapter"]'
+						).textContent;
+				} else {
+					presenceData.details = "Reading a novel";
+					delete presenceData.state;
+				}
+				if (showReading) {
+					presenceData.smallImageKey = "open";
+					presenceData.smallImageText = "Reading";
+				} else {
+					delete presenceData.smallImageKey;
+					delete presenceData.smallImageText;
+				}
+				presenceData.buttons = [
+					{
+						label: "Read Novel",
+						url: location.href
+					}
+				];
+			}
+		}
+	}
+	if (!showButtons) delete presenceData.buttons;
+	if (!showTimestamp) delete presenceData.startTimestamp;
+	browsingTimestamp;
+	if (!showBook) showCover = false;
```

Comment:
And what about this

Reply:
Which ones? @ImRodry Told me to move the deleting portions to the end